core/src/test/java/org/hibernate/ogm/backendtck/id/gen/CustomIdGeneratorTest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.backendtck.id.gen;
+
+import static org.hibernate.ogm.utils.OgmAssertions.assertThat;
+
+import java.util.List;
+
+import org.hibernate.id.IdentifierGenerator;
+import org.hibernate.ogm.utils.jpa.OgmJpaTestCase;
+
+import org.junit.Test;
+
+/**
+ * Test user defined {@link IdentifierGenerator}
+ */
+public class CustomIdGeneratorTest extends OgmJpaTestCase {
+
+	@Test
+	public void test() {
+		inTransaction( em -> {
+			em.persist( new CustomIdGeneratorEntity() );
+		} );
+
+		inTransaction( em -> {
+			List<CustomIdGeneratorEntity> resultList = em.createQuery( "from CustomIdGeneratorEntity" ).getResultList();
+			assertThat( resultList ).hasSize( 1 );
+
+			CustomIdGeneratorEntity entity = resultList.get( 0 );
+			assertThat( entity.id ).isNotEmpty();
```

Comment:
How do you know that the id is generated by your custom generator and not by something else?

We are only inserting one element so I think we can return a constant and check that the id value is the one generated by the custom generator

Reply:
By the way, I know that you have copied the test from the issue, but I don't think that problem was caused by the Random generation of values. 

I also prefer not having random generation in tests, because if it causes a failure we might not be able to identify which value was used.

Reply:
Using a costant also works for the user scenario, because an error should occur when trying to insert two entities with the same id.

Reply:
> Using a costant also works for the user scenario, because an error should occur when trying to insert two entities with the same id.

Thinking again about this, it might not be a good idea to have a constant because some datastore might decide to simply override the existing value because they don't support unique constraints.

I think a better approach would be to create a sequence generator that generates id looking something like: "custom-id-001", "custom-id-002",... 
The test should check that when you insert the first value the id is equal to "custom-id-001"  (and only that value exists in the db).
We can then insert a second value and make sure that the sequence has been incremented as expected (and only those two value exist in the db).